src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseMultiSignMessage.cs

```diff
@@ -49,12 +86,57 @@ public static byte[] SignEmbedded(byte[] embeddedContent, CoseSigner signer, Cos
             => SignCore(detachedContent, null, signer, protectedHeaders, unprotectedHeaders, associatedData, isDetached: true);
 
+
+        /// <summary>
+        /// Signs the specified content and encodes it as a COSE_Sign message with detached content.
+        /// </summary>
+        /// <param name="embeddedContent">The content to sign and to include in the message.</param>
+        /// <param name="signer">The signer information used to sign <paramref name="embeddedContent"/>.</param>
+        /// <param name="protectedHeaders">The protected header parameters to append to the message's content layer.</param>
+        /// <param name="unprotectedHeaders">The unprotected header parameters to append to the message's content layer.</param>
+        /// <param name="associatedData">The extra data associated with the signature, which must also be provided during verification.</param>
+        /// <returns>The encoded message.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="signer"/> is <see langword="null"/>.</exception>
+        /// <exception cref="CryptographicException">Protected and unprotected buckets contain duplicate labels.
+        /// -or-
+        /// One of the specified labels in <see cref="CoseHeaderLabel.CriticalHeaders"/> is missing.</exception>
         public static byte[] SignEmbedded(ReadOnlySpan<byte> embeddedContent, CoseSigner signer, CoseHeaderMap? protectedHeaders = null, CoseHeaderMap? unprotectedHeaders = null, ReadOnlySpan<byte> associatedData = default)
             => SignCore(embeddedContent, null, signer, protectedHeaders, unprotectedHeaders, associatedData, isDetached: false);
 
+        /// <summary>
+        /// Signs the specified content and encodes it as a COSE_Sign message with detached content.
+        /// </summary>
+        /// <param name="detachedContent">The content to sign.</param>
+        /// <param name="signer">The signer information used to sign <paramref name="detachedContent"/>.</param>
+        /// <param name="protectedHeaders">The protected header parameters to append to the message's content layer.</param>
+        /// <param name="unprotectedHeaders">The unprotected header parameters to append to the message's content layer.</param>
+        /// <param name="associatedData">The extra data associated with the signature, which must also be provided during verification.</param>
+        /// <returns>The encoded message.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="detachedContent"/> is <see langword="null"/>.
+        /// -or-
+        /// <paramref name="signer"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="detachedContent"/> does not support reading.
+        /// -or-
+        /// <paramref name="detachedContent"/> does not support seeking.</exception>
+        /// <exception cref="CryptographicException">Protected and unprotected buckets contain duplicate labels.
```

Comment:
In all of these sign operations, the duplicate labels seems like a thing the caller can be smart enough to avoid, or to pre-check, so it's an ArgumentException.

In the case of Encode, where it means a header was added to UnprotectedHeaders that already existed in ProtectedHeaders, that's an InvalidOperationException because the bad state came from something other than a parameter.